docs/proposals/09xx-pluggablitiy/README.md

```diff
@@ -0,0 +1,46 @@
+At this point, dynamic configuration changes are not required. If one would like to change plugins it is needed to update the
+configuration file and restart the EPP pod to initiate a configuration reload.
+- Based on requirements, we might need to allow dynamic configuration changes in the future.  
+
+## Definitions
+- **Inference Extension Framework** - The system created to allow for pluggable extension points in the EPP.
+- **Plugin** - Implementation of framework-defined interface(s) to add or extend logic across the framework.
+
+## Proposal
+
+This proposal aims to define how to structure and setup plugins for different layers of the system. For the sake of the discussion, the proposal focuses on extension points from the scheduling layer, which is the most stable one in terms of pluggability, and from the RequestControl layer.  
+RequestControl might end up having multiple extension points, two of those are already agreed on all parties and will be used as a basis for the discussion:
+
+ - `PreRequest` a plugin that takes the scheduling result and use it to execute logic before the request is sent to the selected endpoint.
+- `PostResponse` - a plugin that is executed after the request was served successfully. The plugin receives the response in order to allow to operate on it. 
+
+Please note that there is no intention to cover all extension points of the system, but only to specify extension points from at least two different layers in order to discuss how to maintain and structure plugins in the repo.
+
+### Plugins maintainace
+Since plugins may implement multiple extension points from different layers of EPP, it makes sense to group plugin implementations in under one directory. 
+For example, instead of keeping plugins under `/pkg/epp/scheduling/framework/plugins`, this proposal suggests to define a shared package to be used for all plugins of the system - `pkg/epp/plugins`.
+
+Additionally, from the same reason, this proposal suggests to have a single Plugin interface across the codebase and all specific plugins should embed this plugin inteface in the extension point. This will allow to register a general `Plugin` instance to EPP and auto-register the plugin in the right extension points (might be very useful when configuration via extrenal configuration file).
+
+### Plugins by code
+
+All extensible structures should be initialized with the plugins in main (under cmd package) and should support setting up extensions by code. Example can be found [here](https://github.com/kubernetes-sigs/gateway-api-inference-extension/blob/main/cmd/epp/main.go#L208-L227).
+
+Additionally, most of the code in the current main file should move to a separate file (may be under separate package) to have a very minimal main.go file which only handles the initialization of plugins and calling a `runner` to run the EPP with the given extensions configuration. This will allow to configure plugins with no code duplication and with minimal code.  
```

Comment:
I think this is a nice to have a not necessarily a requirement. I think what you want to see is that we should have some utilities to facilitate the plugin registration, and by doing so you end up with minimal code in main.go. But simplifying main.go isn't the goal of this proposal

So what I suggest is just describing that requirement for that utility package , and don't mention about how to simplify main.go (it's just a result of defining that utility package)/